source/extensions/compression/common/decompressor/factory_base.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "envoy/compression/decompressor/config.h"
+#include "envoy/compression/decompressor/decompressor.h"
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Common {
+namespace Decompressor {
+
+template <class ConfigProto>
+class DecompressorLibraryFactoryBase
+    : public Envoy::Compression::Decompressor::NamedDecompressorLibraryConfigFactory {
+public:
+  Envoy::Compression::Decompressor::DecompressorFactoryPtr
+  createDecompressorLibraryFromProto(const Protobuf::Message& proto_config,
```

Comment:
I know we had talked about this offline, but `createDecompressorLibraryFromProto` might be a little confusing since it actually returns a factory. Would it make sense to rename this function (and the corresponding mirror for compression) to something like `createDecompressorFactoryFromProto`? Could also be a todo so we can change compression/decompression at the same time